Xamarin.Forms.Core/BindableObject.cs

```diff
@@ -240,7 +240,7 @@ public static void SetInheritedBindingContext(BindableObject bindable, object va
 			}
 			else
 			{
-				bindable._inheritedContext = value;
+				bindable._inheritedContext = new WeakReference(value);
```

Comment:
As a UNO instead of creating a new WeakReference instance, you can use code like this to reduce GC pressure:

```suggestion
				bindable._inheritedContext ??= new WeakReference(null);
				bindable._inheritedContext.Target = value;
```